### DISCUSSION TITLE
Font larger than 96 point?

### DISCUSSION THREAD

**Discussion by allanyates** [Bodmer/TFT_eSPI#1966] [Q&A - Fonts]
Library is working beautifully with the 3.5" TFT LCD 320x480 from Adafruit (HX8357D).

I'm just wondering on the best method for obtaining a large point full resolution font? I can scale the provided fonts with setTextSize, but they get very blocky. Using, http://oleddisplay.squix.ch, I was able to generate the very nice 96 point font in the attached example. However, above this size, that utility seems to wrap and generate negative numbers that don't compile into a uint.

My objective is to about double the size of that top row of text (clock digits), such that it about fills the top seciont of the display. The red rectangles and just so I can see the size of the current sprites.

Thanks,

Allan.

![IMG_7282](https://user-images.githubusercontent.com/70734610/185391627-30bce9ef-aa96-423f-930c-8608bbdc85b8.JPG)

**Comment by Bodmer** [comment]
The Adafruit font uses an int8_t so sets a limit of 127 for the height above the baseline, and 127 below. This is a limitation of that font format not the library sp this limits the maximum size to around 95 points.

Options are to use a different font format that accepts larger characters:
  Smooth fonts (anti-aliased)
  or this library https://github.com/Bodmer/U8g2_for_TFT_eSPI
  of this library https://github.com/Bodmer/TFT_eSPI_ext

**Reply by Bodmer** [reply]
Here's an example large smooth font (240pt). Only numbers and : are in the font!
[Large_font_1966.zip](https://github.com/Bodmer/TFT_eSPI/files/9374823/Large_font_1966.zip)

**Reply by musseilm** [reply]
@Bodmer Could you explain how you create a Font with only numbers? I do not see this option in Processing IDE.

This could be very helpful for me.
Thanks

**Reply by Bodmer** [reply]
Run the processing sketch here:
https://github.com/Bodmer/TFT_eSPI/tree/master/Tools/Create_Smooth_Font/Create_font

Read the comments about range selection.

**Comment by allanyates** [comment]
Brilliant! The your last example got me going. I decided something just a little small was needed. I found your Processing sketch for font creation, https://github.com/Bodmer/TFT_eSPI/blob/master/Tools/Create_Smooth_Font/Create_font/Create_font.pde, and then the referenced file converter, https://tomeko.net/online_tools/file_to_hex.php?lang=en. This all worked perfectly. Many thanks as always.

My biggest surprise was how much space these fonts can consume. This large font, limited to the 0-9 : characters, took 116 kBytes of program space on the ESP32.

![IMG_7287](https://user-images.githubusercontent.com/70734610/185480076-c14d0b5d-bf4e-4c42-a7d7-224893670e59.JPG)

**Comment by XFer012** [comment]
> Using, http://oleddisplay.squix.ch, I was able to generate the very nice 96 point font in the attached example

Hello, please, could you give a few directions as to how to adapt the output of that online converter to a proper TFT_eSPI header file?
As is, it is not compatible it seems to me.

Thanks!
